lib/spack/spack/build_systems/python.py

```diff
@@ -399,3 +399,10 @@ def remove_files_from_view(self, view, merge_map):
                 view.remove_file(src, dst)
             else:
                 os.remove(dst)
+
+    @property
+    def site_packages_dir(self):
```

Comment:
Will this attribute only be present for `PythonPackage` and `clingo`? I would much rather it be present for all Python extensions. I don't see why we need to do anything in `build_systems`.

Reply:
See the failure here: https://github.com/spack/spack/pull/25138/checks?check_run_id=3223289393 Can we keep in the Python class for the time being and fix it more permanently in a following PR?

Reply:
Won't that cause issues for all packages that `extend` Python but don't subclass `PythonPackage`? I don't want to break those packages with this PR.

Reply:
It won't cause any issue I think, since this PR is adding a method not substituting one. They just won't have a package level `site_packages_dir` attribute, which is an inconsistency, but I think can be solved later.

Reply:
But don't we try to use this in `setup_dependent_run_environment` for all Python packages? Won't that crash for those examples?

Reply:
Closing as outdated